test/parallel/test-whatwg-encoding-textencoder.js

```diff
@@ -2,50 +2,93 @@
 'use strict';
 
 const common = require('../common');
-
 const assert = require('assert');
 const { TextDecoder, TextEncoder } = require('util');
 const { customInspectSymbol: inspect } = require('internal/util');
 
-const encoded = Buffer.from([0xef, 0xbb, 0xbf, 0x74, 0x65,
-                             0x73, 0x74, 0xe2, 0x82, 0xac]);
+// This test checks TextEncoder's instantiation, default encoding,
+// encoding capabilities, and conformity with util.inspect().
 
-// Make Sure TextEncoder exists
-assert(TextEncoder);
+{
+  const textEncoder = new TextEncoder();
+  assert.ok(textEncoder);
+  assert.strictEqual(textEncoder instanceof TextEncoder, true);
```

Comment:
This block checks `TextEncoder`'s instantiation. It's similar to line `20 (pre-change)`. However, it also performs a type-check. It's inane though.

Suppose `const textEncoder = new TextEncoder()` on line `13 (post-change)` executed without throwing an error. How could `textEncoder` **not** be an `instanceof` `TextEncoder`? 

I'm not sure. Oh well. I didn't think it'd hurt to include the type-check.

Reply:
> How could `textEncoder` **not** be an `instanceof` `TextEncoder`? 

```js
function WeirdClass() {
  // return an object that doesn't have WeirdClass.prototype as its prototype
  return {};
}
assert.strictEqual(new WeirdClass() instanceof WeirdClass, false);
```

JavaScript is very weird indeed.

> I didn't think it'd hurt to include the type-check. So, I included it. It provides a bit more satisfaction, I suppose.

I'd say doing so is unnecessary. If the instance does not pass this type check, something is likely to be very wrong, and thus unlikely to make it pass all the other tests we have.

Reply:
The same principle could apply to @starkwang's comment. Yet the conclusions are different.

If this type check fails, subsequent tests will likely fail.
If `assert(TextEncoder)` fails, subsequent tests will likely fail.

They both could be considered unnecessary. But we're treating them differently. We discard this type check and keep `assert(TextEncoder)`? I don't understand.

Reply:
I’m fine with removing the `assert(TextEncoder)`. In fact I don’t think such a test exists anywhere else.

P.S. there are cases where `if (a)` gives false, but `a()` still works. Or rather, there is one case where that is true: `document.all`. Check out the second Note in the [spec](https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#the-htmlallcollection-interface) — it really is quite interesting. But I digress.

Reply:
It's interesting, but I feel like that's going to cause someone massive headaches someday.